// nightwatch/DiscoveryNow_WebUI_TestCases/tests/01_discoverNow.js
/*
 * DiscoveryNow Web UI
 * Login --> Create Topic --> Delete Topic
 * 11 Test Cases
 *
 * Attensity Group LLC (2015)
 */
 
    qcredentials = require('./qcredentials.json');
	qconfig = require('./qconfig.json');
		
	var qaURL = qconfig.qaURL;
	var loginButton = qconfig.loginButton;
	var newTopic = qconfig.newTopic;
	var sources = qconfig.sources;
	var allSources = qconfig.allSources;
	var doneButton = qconfig.doneButton;
	var saveTopic = qconfig.saveTopic;
	var home = qconfig.home;
	var topicName1 = qconfig.topicName1;
	var topicName2 = qconfig.topicName2;
	var topicDeleteButton = qconfig.topicDeleteButton;
	var topicDeleteModel = qconfig.topicDeleteModel;
	var topicDeleteCancelButton = qconfig.topicDeleteCancelButton;
	var topicDeleteDeleteButton = qconfig.topicDeleteDeleteButton;
	
exports.url=function(Url) {
	Url
	  .url(qaURL)
	  .pause(3000)
      .waitForElementVisible('body', 9000)
	  .maximizeWindow()
	  .pause(3000)
};

exports.login=function(client) {
	client
	  .setValue('input#account', qcredentials.account)
      .setValue('input#login', qcredentials.login)
	  .setValue('input#password', qcredentials.password)
	  .pause(1000)
	  .click(loginButton)
	  .pause(3000)
};

exports.newTopic=function(client) {
	client
	  .click(newTopic)
	  .pause(3000)
	  .setValue('input[type=text]', ['super'])
	  .pause(3000)
	  .keys([client.Keys.SPACE])
	  .pause(2000)
};

exports.sources=function(client) {
	client
	  .click(sources)
	  .pause(2000)
};

exports.allSources=function(client) {
	client
	  .click(allSources)
	  .pause(2000)
	  .click(doneButton)
	  .pause(10000)
};

exports.saveTopic=function(client) {
	client
	  .click(saveTopic)
	  .pause(3000)
	  .setValue('input[name=topicName]', ['AutomationWebUITopic'])
	  .pause(3000)
	  .keys([client.Keys.TAB])
	  .keys([client.Keys.TAB])
	  .keys([client.Keys.TAB])
	  .keys([client.Keys.TAB])
	  .keys([client.Keys.ENTER])
	  .pause(10000)
};

exports.home=function(client) {
	client
	  .click(home)
	  .pause(5000)
};

exports.topicName1=function(client) {
	client
	  .useXpath()
	  .moveToElement(topicName1,10, 10)
	  .pause(3000)
};

exports.topicDeleteCancel=function(client) {
	client
	  .click(topicDeleteButton)
	  .pause(5000)
	  .useCss()
	  .moveToElement(topicDeleteModel,10, 10)
	  .pause(3000)
	  .click(topicDeleteCancelButton)
	  .pause(5000)
};

exports.topicName1a=function(client) {
	client
	  .useXpath()
	  .moveToElement(topicName1,10, 10)
	  .pause(3000)
};

exports.topicDeleteDelete=function(client) {
	client
	  .click(topicDeleteButton)
	  .pause(5000)
	  .useCss()
	  .moveToElement(topicDeleteModel,10, 10)
	  .pause(3000)
	  .click(topicDeleteDeleteButton)
	  .pause(5000)
};

exports.end=function(client) {
	client
	  .end();
};
