// nightwatch/DiscoveryNow_WebUI_TestCases/tests/02_discoverNow.js
/*
 * DiscoveryNow Web UI
 * Login --> Create Topic --> Company Profile --> Delete Topic
 * 
 *
 * Attensity Group LLC (2015)
 */
 
    qcredentials = require('./qcredentials.json');
	qconfig = require('./qconfig.json');
		
	var qaURL = qconfig.qaURL;
	var loginButton = qconfig.loginButton;
	var settings = qconfig.settings;
	var companyProfile = qconfig.companyProfile;
	var showGuidedAssistanceOn = qconfig.showGuidedAssistanceOn;
	var showGuidedAssistanceOff = qconfig.showGuidedAssistanceOff;
	
exports.url=function(client) {
	client
	  .url(qaURL)
	  .pause(3000)
      .waitForElementVisible('body', 9000)
	  .maximizeWindow()
	  .pause(3000)
};

exports.login=function(client) {
	client
	  .setValue('input#account', qcredentials.account)
      .setValue('input#login', qcredentials.login)
	  .setValue('input#password', qcredentials.password)
	  .pause(1000)
	  .click(loginButton)
	  .pause(3000)
};

exports.settings=function(client) {
	client
	  .click(settings)
	  .pause(2000)	
};

exports.CompanyProfile=function(client) {
	client
	  .useCss()
	  .click(companyProfile)
	  .pause(3000)
	  //Show guided assistance//
	  .useXpath()
	  .click(showGuidedAssistanceOn)
	  .pause(2000)
	  .click(showGuidedAssistanceOff)
	  .pause(2000)	
	  //Company://
	  .click('/html/body/div[1]/div[2]/div/div/div/div[3]/div/div[2]/topic-tag-input/div/div/div[3]/topic-tag/div/div[3]/a')
	  .click('/html/body/div[1]/div[2]/div/div/div/div[3]/div/div[2]/topic-tag-input/div/div/div[3]/input')
	  .pause(1000)
	  .useCss().setValue('input[type=text]', ['Verizon'])
	  .pause(1000)
	  .useXpath().click('/html/body/div[1]/div[2]/div/div/div/div[3]/div/div[2]/topic-tag-input/div/div/div[5]/ul/li[1]')
	  .pause(1000)
 	  //Brands://
	  .click('/html/body/div[1]/div[2]/div/div/div/div[3]/div/div[4]/topic-tag-input/div/div/div[3]/topic-tag/div/div[3]/a')
	  .click('/html/body/div[1]/div[2]/div/div/div/div[3]/div/div[4]/topic-tag-input/div/div/div[3]/input')
	  .pause(1000)
	  .setValue('/html/body/div[1]/div[2]/div/div/div/div[3]/div/div[4]/topic-tag-input/div/div/div[3]/input', ['Android'])
	  .click('/html/body/div/div[2]/div/div/div/div[3]/div/div[4]/topic-tag-input/div/div/div[5]/ul')
	  .keys([client.Keys.ENTER])
	  .pause(1000)
 	  //Products://
	  .click('/html/body/div[1]/div[2]/div/div/div/div[3]/div/div[6]/topic-tag-input/div/div/div[3]/topic-tag/div/div[3]/a')
	  .click('/html/body/div/div[2]/div/div/div/div[3]/div/div[6]/topic-tag-input/div/div/div[3]/input')
	  .pause(1000)
	  .setValue('/html/body/div/div[2]/div/div/div/div[3]/div/div[6]/topic-tag-input/div/div/div[3]/input', ['Smartphone'])
	  .click('/html/body/div[1]/div[2]/div/div/div/div[3]/div/div[6]/topic-tag-input/div/div/div[5]/ul')
	  .keys([client.Keys.ENTER])
	  .pause(1000)
	  //Competitors://
	  .click('/html/body/div[1]/div[2]/div/div/div/div[3]/div/div[8]/topic-tag-input/div/div/div[3]/topic-tag/div/div[3]/a')
	  .click('/html/body/div[1]/div[2]/div/div/div/div[3]/div/div[8]/topic-tag-input/div/div/div[3]/input')
	  .pause(1000)
	  .setValue('/html/body/div[1]/div[2]/div/div/div/div[3]/div/div[8]/topic-tag-input/div/div/div[3]/input', ['Sprint'])
	  .click('/html/body/div[1]/div[2]/div/div/div/div[3]/div/div[8]/topic-tag-input/div/div/div[5]/ul')
	  .keys([client.Keys.ENTER])
	  .pause(1000)
	  //Competitors Brands://
	  .click('/html/body/div[1]/div[2]/div/div/div/div[3]/div/div[10]/topic-tag-input/div/div/div[3]/topic-tag/div/div[3]/a')
	  .click('/html/body/div[1]/div[2]/div/div/div/div[3]/div/div[10]/topic-tag-input/div/div/div[3]/input')
	  .pause(1000)
	  .setValue('/html/body/div[1]/div[2]/div/div/div/div[3]/div/div[10]/topic-tag-input/div/div/div[3]/input', ['iphone'])
	  .click('/html/body/div[1]/div[2]/div/div/div/div[3]/div/div[10]/topic-tag-input/div/div/div[5]/ul')
	  .keys([client.Keys.ENTER])
	  .pause(1000)
	  //Competitors Products://
	  .click('/html/body/div[1]/div[2]/div/div/div/div[3]/div/div[12]/topic-tag-input/div/div/div[3]/topic-tag/div/div[3]/a')
	  .click('/html/body/div[1]/div[2]/div/div/div/div[3]/div/div[12]/topic-tag-input/div/div/div[3]/input')
	  .pause(1000)
	  .setValue('/html/body/div[1]/div[2]/div/div/div/div[3]/div/div[12]/topic-tag-input/div/div/div[3]/input', ['Smartphone'])
	  .click('/html/body/div[1]/div[2]/div/div/div/div[3]/div/div[12]/topic-tag-input/div/div/div[5]/ul')
	  .keys([client.Keys.ENTER])
	  .pause(1000)
 };
	  
exports.CompanyProfileMoreFields=function(client) {
	client
	  .useXpath()
	  .click('/html/body/div[1]/div[2]/div/div/div/div[3]/div/a')
	  .pause(1000)
	  //Services://
	  .click('/html/body/div[1]/div[2]/div/div/div/div[3]/div/div[13]/div/div[2]/topic-tag-input/div/div/div[3]/topic-tag/div/div[3]/a')
	  .click('/html/body/div[1]/div[2]/div/div/div/div[3]/div/div[13]/div/div[2]/topic-tag-input/div/div/div[3]/input')
	  .pause(1000)
	  .setValue('/html/body/div[1]/div[2]/div/div/div/div[3]/div/div[13]/div/div[2]/topic-tag-input/div/div/div[3]/input', ['Cellular'])
	  .click('/html/body/div[1]/div[2]/div/div/div/div[3]/div/div[13]/div/div[2]/topic-tag-input/div/div/div[5]/ul')
	  .keys([client.Keys.ENTER])
	  .pause(1000)
	  //People://
	  .click('/html/body/div[1]/div[2]/div/div/div/div[3]/div/div[13]/div/div[4]/topic-tag-input/div/div/div[3]/topic-tag/div/div[3]/a')
	  .click('/html/body/div[1]/div[2]/div/div/div/div[3]/div/div[13]/div/div[4]/topic-tag-input/div/div/div[3]/input')
	  .pause(1000)
	  .setValue('/html/body/div[1]/div[2]/div/div/div/div[3]/div/div[13]/div/div[4]/topic-tag-input/div/div/div[3]/input', ['VIP'])
	  .click('/html/body/div[1]/div[2]/div/div/div/div[3]/div/div[13]/div/div[4]/topic-tag-input/div/div/div[5]/ul')
	  .keys([client.Keys.ENTER])
	  .pause(1000)
	  //Places://
	  .click('/html/body/div[1]/div[2]/div/div/div/div[3]/div/div[13]/div/div[6]/topic-tag-input/div/div/div[3]/topic-tag/div/div[3]/a')
	  .click('/html/body/div[1]/div[2]/div/div/div/div[3]/div/div[13]/div/div[6]/topic-tag-input/div/div/div[3]/input')
	  .pause(1000)
	  .setValue('/html/body/div[1]/div[2]/div/div/div/div[3]/div/div[13]/div/div[6]/topic-tag-input/div/div/div[3]/input', ['"United States"'])
	  .click('/html/body/div[1]/div[2]/div/div/div/div[3]/div/div[13]/div/div[6]/topic-tag-input/div/div/div[5]/ul')
	  .keys([client.Keys.ENTER])
	  .pause(1000)
	  //Industry Terms://
	  .click('/html/body/div[1]/div[2]/div/div/div/div[3]/div/div[13]/div/div[8]/topic-tag-input/div/div/div[3]/topic-tag/div/div[3]/a')
	  .click('/html/body/div[1]/div[2]/div/div/div/div[3]/div/div[13]/div/div[8]/topic-tag-input/div/div/div[3]/input')
	  .pause(1000)
	  .setValue('/html/body/div[1]/div[2]/div/div/div/div[3]/div/div[13]/div/div[8]/topic-tag-input/div/div/div[3]/input', ['Device'])
	  .click('/html/body/div[1]/div[2]/div/div/div/div[3]/div/div[13]/div/div[8]/topic-tag-input/div/div/div[5]/ul')
	  .keys([client.Keys.ENTER])
	  .pause(1000)
};	  
	  
exports.CompanyProfileSave=function(client) {
	client
	  .useXpath()
	  .click('/html/body/div[1]/div[2]/div/div/div/div[1]/div/div[3]/div/button[2]')
	  .pause(2000)
	  .assert.containsText('/html/body/div[1]/div[2]/div/div/div/div[3]/div[1]/div/span', 'Company Profile Has Been Saved')
	  .pause(2000)
};

exports.logout=function(client) {
	client
	  .useXpath()
	  .click('//*[@id="navbar"]/ul[2]/li[2]/a')
	  .pause(2000)
	  .click('//*[@id="navbar"]/ul[2]/li[2]/ul/li[4]/a')
	  .pause(2000)
};

exports.end=function(client) {
	client
	  .end(); 
};
