// nightwatch/DiscoveryNow_WebUI_TestCases/tests/03_discoverNow.js
/*
 * DiscoveryNow Web UI
 * Login --> Create Topic --> Filters --> Delete Topic
 * 11 Test Cases
 *
 * Attensity Group LLC (2015)
 */
 
    qcredentials = require('./qcredentials.json');
	qconfig = require('./qconfig.json');
		
	var qaURL = qconfig.qaURL;
	var loginButton = qconfig.loginButton;
	var newTopic = qconfig.newTopic;
	var sources = qconfig.sources;
	var allSources = qconfig.allSources;
	var doneButton = qconfig.doneButton;
	var saveTopic = qconfig.saveTopic;
	var home = qconfig.home;
	var topicName1 = qconfig.topicName1;
	var topicName2 = qconfig.topicName2;
	var topicDeleteButton = qconfig.topicDeleteButton;
	var topicDeleteModel = qconfig.topicDeleteModel;
	var topicDeleteCancelButton = qconfig.topicDeleteCancelButton;
	var topicDeleteDeleteButton = qconfig.topicDeleteDeleteButton;
	
exports.url=function(client) {
	client
	  .url(qaURL)
	  .pause(3000)
      .waitForElementVisible('body', 9000)
	  .maximizeWindow()
	  .pause(3000)
};

exports.login=function(client) {
	client
	  .setValue('input#account', qcredentials.account)
      .setValue('input#login', qcredentials.login)
	  .setValue('input#password', qcredentials.password)
	  .pause(1000)
	  .click(loginButton)
	  .pause(3000)
};

exports.newTopic=function(client) {
	client
	  .click(newTopic)
	  .pause(3000)
	  .setValue('input[type=text]', ['super'])
	  .pause(3000)
	  .keys([client.Keys.SPACE])
	  .pause(2000)
};

exports.sources=function(client) {
	client
	  .click(sources)
	  .pause(2000)
};

exports.allSources=function(client) {
	client
	  .click(allSources)
	  .pause(2000)
	  .click(doneButton)
	  .pause(10000)
};

exports.saveTopic=function(client) {
	client
	  .click(saveTopic)
	  .pause(3000)
	  .setValue('input[name=topicName]', ['AutomationWebUITopic'])
	  .pause(3000)
	  .keys([client.Keys.TAB])
	  .keys([client.Keys.TAB])
	  .keys([client.Keys.TAB])
	  .keys([client.Keys.TAB])
	  .keys([client.Keys.ENTER])
	  .pause(10000)
};

exports.filterTimePeriod=function(client) {
	client
	  .useXpath()
	  .click('//*[@id="att-dashboard-filter-container"]/div/accordion/div/div[1]/div[1]/h4/a/span')
	  .pause(1000)
	  .click('//*[@id="att-dashboard-filter-container"]/div/accordion/div/div[1]/div[2]/div/date-filter/div/form/div[1]/select/option[4]')
	  .pause(1000)
	  .click('//*[@id="att-dashboard-filter-container"]/div/accordion/div/div[1]/div[2]/div/date-filter/div/form/div[1]/select/option[1]')
	  .pause(1000)
	  .click('//*[@id="att-dashboard-filter-container"]/div/accordion/div/div[1]/div[2]/div/date-filter/div/form/div[1]/select/option[3]')
	  .pause(1000)
	  .click('//*[@id="att-dashboard-filter-container"]/div/accordion/div/div[1]/div[2]/div/date-filter/div/form/div[1]/select/option[2]')
	  .pause(1000)
	  .click('//*[@id="att-dashboard-filter-container"]/div/accordion/div/div[1]/div[2]/div/date-filter/div/form/div[1]/select/option[1]')
};

exports.filterSentiment=function(client) {
	client
	  .useXpath()
	  .click('//*[@id="att-dashboard-filter-container"]/div/accordion/div/div[4]/div[1]/h4/a/span')
	  .pause(1000)
	  .click('//*[@id="att-dashboard-filter-container"]/div/accordion/div/div[4]/div[2]/div/sentiment-filter/div/div[1]/div/button')
	  .pause(1000)
	  .click('//*[@id="att-dashboard-filter-container"]/div/accordion/div/div[4]/div[2]/div/sentiment-filter/div/div[2]/div/button')
	  .pause(1000)
	  .click('//*[@id="att-dashboard-filter-container"]/div/accordion/div/div[4]/div[2]/div/sentiment-filter/div/div[3]/div/button')
	  .pause(1000)
	  .click('//*[@id="att-dashboard-filter-container"]/div/accordion/div/div[4]/div[2]/div/sentiment-filter/div/div[4]/div/button')
	  .pause(1000)
	  .click('//*[@id="att-dashboard-filter-container"]/div/accordion/div/div[4]/div[1]/h4/a/button')
	  .pause(1000)
	  .click('//*[@id="att-dashboard-filter-container"]/div/accordion/div/div[4]/div[1]/h4/a/span')
	  .pause(1000)
};

exports.filterGender=function(client) {
	client
	  .useXpath()
	  .click('//*[@id="att-dashboard-filter-container"]/div/accordion/div/div[6]/div[1]/h4/a')
	  .pause(1000)
	  .click('//*[@id="att-dashboard-filter-container"]/div/accordion/div/div[6]/div[2]/div/gender-filter/div[2]/div[1]/a')
	  .pause(1000)
	  .click('//*[@id="att-dashboard-filter-container"]/div/accordion/div/div[6]/div[2]/div/gender-filter/div[2]/div[2]/a')
	  .pause(1000)
	  .click('//*[@id="att-dashboard-filter-container"]/div/accordion/div/div[6]/div[1]/h4/a/button')
	  .pause(1000)
	  .click('//*[@id="att-dashboard-filter-container"]/div/accordion/div/div[6]/div[1]/h4/a')
	  .pause(1000)
};

exports.filterSources=function(client) {
	client
	  .useXpath()
	  .click('//*[@id="att-dashboard-filter-container"]/div/accordion/div/div[7]/div[1]/h4/a/span')
	  .pause(1000)
	  .click('//*[@id="att-dashboard-filter-container"]/div/accordion/div/div[7]/div[2]/div/sources-filter/div/div[1]/div/button')
	  .pause(1000)
	  .click('//*[@id="att-dashboard-filter-container"]/div/accordion/div/div[7]/div[2]/div/sources-filter/div/div[2]/div/button')
	  .pause(1000)
	  .click('//*[@id="att-dashboard-filter-container"]/div/accordion/div/div[7]/div[2]/div/sources-filter/div/div[3]/div/button')
	  .pause(1000)
	  .click('//*[@id="att-dashboard-filter-container"]/div/accordion/div/div[7]/div[2]/div/sources-filter/div/div[4]/div/button')
	  .pause(1000)
	  .click('//*[@id="att-dashboard-filter-container"]/div/accordion/div/div[7]/div[2]/div/sources-filter/div/div[5]/div/button')
	  .pause(1000)
	  .click('//*[@id="att-dashboard-filter-container"]/div/accordion/div/div[7]/div[2]/div/sources-filter/div/div[6]/div/button')
	  .pause(1000)
	  .click('//*[@id="att-dashboard-filter-container"]/div/accordion/div/div[7]/div[2]/div/sources-filter/div/div[7]/div/button')
	  .pause(1000)
	  .click('//*[@id="att-dashboard-filter-container"]/div/accordion/div/div[7]/div[2]/div/sources-filter/div/div[8]/div/button')
	  .pause(1000)
	  .click('//*[@id="att-dashboard-filter-container"]/div/accordion/div/div[7]/div[1]/h4/a/button')
	  .pause(1000)
	  .click('//*[@id="att-dashboard-filter-container"]/div/accordion/div/div[7]/div[1]/h4/a/span')
	  .pause(1000)
};

exports.filterInfluencers=function(client) {
	client
	  .useXpath()
	  .click('//*[@id="att-dashboard-filter-container"]/div/accordion/div/div[8]/div[1]/h4/a')
	  .pause(1000)
	  .click('//*[@id="att-dashboard-filter-container"]/div/accordion/div/div[8]/div[2]/div/influencer-metrics-filter/div/div[1]/a')
	  .pause(1000)
	  .click('//*[@id="att-dashboard-filter-container"]/div/accordion/div/div[8]/div[2]/div/influencer-metrics-filter/div/div[1]/div/div/div/span[2]')
	  .pause(1000)
	  .click('//*[@id="att-dashboard-filter-container"]/div/accordion/div/div[8]/div[2]/div/influencer-metrics-filter/div/div[2]/a')
	  .pause(1000)
	  .click('//*[@id="att-dashboard-filter-container"]/div/accordion/div/div[8]/div[2]/div/influencer-metrics-filter/div/div[2]/div/div/div/span[2]')
	  .pause(1000)
	  .click('//*[@id="att-dashboard-filter-container"]/div/accordion/div/div[8]/div[1]/h4/a/button')
	  .pause(1000)
	  .click('//*[@id="att-dashboard-filter-container"]/div/accordion/div/div[8]/div[1]/h4/a')
	  .pause(1000)
};

exports.filterCompanyProfile=function(client) {
	client
	  .useXpath()
	  .click('//*[@id="att-dashboard-filter-container"]/div/accordion/div/div[9]/div[1]/h4/a/span')
	  .pause(1000)
	  .click('//*[@id="0"]')
	  .pause(1000)
	  .click('//*[@id="1"]')
	  .pause(1000)
	  .click('//*[@id="2"]')
	  .pause(1000)
	  .click('//*[@id="3"]')
	  .pause(1000)
	  .click('//*[@id="4"]')
	  .pause(1000)
	  .click('//*[@id="5"]')
	  .pause(1000)
	  .click('//*[@id="6"]')
	  .pause(1000)
	  .click('//*[@id="7"]')
	  .pause(1000)
	  .click('//*[@id="8"]')
	  .pause(1000)
	  .click('//*[@id="9"]')
	  .pause(1000)
	  .click('//*[@id="att-dashboard-filter-container"]/div/accordion/div/div[9]/div[1]/h4/a/button')
	  .pause(1000)
};

exports.home=function(client) {
	client
	  .useCss()
	  .click(home)
	  .pause(3000)
};

exports.topicName1=function(client) {
	client
	  .useXpath()
	  .moveToElement(topicName1,10, 10)
	  .pause(3000)
};

exports.topicDeleteCancel=function(client) {
	client
	  .useXpath()
	  .click(topicDeleteButton)
	  .pause(3000)
	  .useCss()
	  .moveToElement(topicDeleteModel,10, 10)
	  .pause(3000)
	  .click(topicDeleteCancelButton)
	  .pause(5000)
};

exports.topicName1a=function(client) {
	client
	  .useXpath()
	  .moveToElement(topicName1,10, 10)
	  .pause(3000)
};

exports.topicDeleteDelete=function(client) {
	client
	  .useXpath()
	  .click(topicDeleteButton)
	  .pause(3000)
	  .useCss()
	  .moveToElement(topicDeleteModel,10, 10)
	  .pause(3000)
	  .click(topicDeleteDeleteButton)
	  .pause(5000)
};

exports.end=function(client) {
	client
	  .end(); 
};
