// nightwatch/DiscoveryNow_WebUI_TestCases/tests/04_discoverNow.js
/*
 * DiscoveryNow Web UI
 * Login --> Create Topic --> Navigate Dashboards --> Alerts Settings --> Settings
 * 
 *
 * Attensity Group LLC (2015)
 */
 
    qcredentials = require('./qcredentials.json');
	qconfig = require('./qconfig.json');
		
	var qaURL = qconfig.qaURL;
	var loginButton = qconfig.loginButton;
	var newTopic = qconfig.newTopic;
	var sources = qconfig.sources;
	var allSources = qconfig.allSources;
	var doneButton = qconfig.doneButton;
	var saveTopic = qconfig.saveTopic;
	var home = qconfig.home;
	var topicName1 = qconfig.topicName1;
	var topicName2 = qconfig.topicName2;
	var topicDeleteButton = qconfig.topicDeleteButton;
	var topicDeleteModel = qconfig.topicDeleteModel;
	var topicDeleteCancelButton = qconfig.topicDeleteCancelButton;
	var topicDeleteDeleteButton = qconfig.topicDeleteDeleteButton;
	
exports.url=function(client) {
	client
	  .url(qaURL)
	  .pause(3000)
      .waitForElementVisible('body', 9000)
	  .maximizeWindow()
	  .pause(3000)
};

exports.login=function(client) {
	client
	  .setValue('input#account', qcredentials.account)
      .setValue('input#login', qcredentials.login)
	  .setValue('input#password', qcredentials.password)
	  .pause(1000)
	  .click(loginButton)
	  .pause(3000)
};

exports.newTopic=function(client) {
	client
	  .click(newTopic)
	  .pause(3000)
	  .setValue('input[type=text]', ['super'])
	  .pause(3000)
	  .keys([client.Keys.SPACE])
	  .pause(2000)
};

exports.sources=function(client) {
	client
	  .click(sources)
	  .pause(2000)
};

exports.allSources=function(client) {
	client
	  .click(allSources)
	  .pause(2000)
	  .click(doneButton)
	  .pause(10000)
};

exports.saveTopic=function(client) {
	client
	  .click(saveTopic)
	  .pause(3000)
	  .setValue('input[name=topicName]', ['AutomationWebUITopic'])
	  .pause(3000)
	  .keys([client.Keys.TAB])
	  .keys([client.Keys.TAB])
	  .keys([client.Keys.TAB])
	  .keys([client.Keys.TAB])
	  .keys([client.Keys.ENTER])
	  .pause(10000)
};

exports.navMenu=function(client) {
	client
	 .useXpath()
	 //Dataset Overview//
	.click('//*[@id="topics"]/li[1]/a')
	.pause(3000)
	 //Stream//
	.click('//*[@id="topics"]/li[2]/a')
	.pause(3000)
	 //Top Categories//
	.click('//*[@id="topics"]/li[3]/a')
	.pause(3000)
	 //Trending//
	.click('//*[@id="topics"]/li[4]/a')
	.pause(3000)
	 //Company Sentiment//
	.click('//*[@id="topics"]/li[5]/a')
	.pause(3000)
};	
	
exports.alertSettings=function(client) {
	client
	 .useXpath()	
	 //Alerts Settings//
	.click('//*[@id="topics"]/li[7]/a')
	.pause(5000)
	 //Volume//	
	.click('//*[@id="att-alerts-settings"]/div/div[2]/div[2]/div[2]/div/div[1]/div[1]/label/span')
	 //Spikes//
	.click('//*[@id="att-alerts-settings"]/div/div[2]/div[2]/div[3]/div/div[1]/div[1]/label/span')
	 //Followers//
	.click('//*[@id="att-alerts-settings"]/div/div[2]/div[2]/div[4]/div/div[2]/div/label/span')
	 //Klout//
	.click('//*[@id="att-alerts-settings"]/div/div[2]/div[2]/div[4]/div/div[4]/div/label/span[1]')
	 //Sentiment//
	.click('//*[@id="att-alerts-settings"]/div/div[2]/div[2]/div[5]/div/div[1]/div[1]/label/span')
	.pause(3000)
	 //Volume Threshold//
	.click('//*[@id="att-alerts-settings"]/div/div[2]/div[2]/div[2]/div/div[2]/div/div/span[3]')
	 //Spikes Threshold//
	.click('//*[@id="att-alerts-settings"]/div/div[2]/div[2]/div[3]/div/div[2]/div/div/span[3]')
	 //Followers Threshold//
	.click('//*[@id="att-alerts-settings"]/div/div[2]/div[2]/div[4]/div/div[3]/div/div/span[3]')
	 //Klout Threshold//
	.click('//*[@id="att-alerts-settings"]/div/div[2]/div[2]/div[4]/div/div[5]/div/div/span[3]')
	 //Sentiment Threshold//
	.click('//*[@id="att-alerts-settings"]/div/div[2]/div[2]/div[5]/div/div[2]/div/div/span[3]')
};

exports.alerts=function(client) {
	client
	 .useXpath()
	 //Alerts//
	.click('//*[@id="menu-content"]/li[3]')
	.pause(3000)
	 //Filter Alerts//
	.click('/html/body/div[1]/div[2]/div/div/div/div/div[1]/div[2]/div/button')
	.pause(3000)
	 //Account Alerts//
	.click('/html/body/div[1]/div[2]/div/div/div/div/div[1]/div[2]/div/ul/li[4]')
	.pause(3000)
	 //Filter Alerts//
	.click('/html/body/div[1]/div[2]/div/div/div/div/div[1]/div[2]/div/button')
	.pause(3000)
	 //Topic Alerts//
	.click('/html/body/div[1]/div[2]/div/div/div/div/div[1]/div[2]/div/ul/li[5]')
	.pause(3000)
	 //All Alerts//
	.click('/html/body/div[1]/div[2]/div/div/div/div/div[1]/div[2]/div/ul/li[1]')
	.pause(3000)
	 //Dismiss All//
	.click('/html/body/div[1]/div[2]/div/div/div/div/div[1]/div[3]/span/button')
	.pause(5000)
};

exports.settings=function(client) {
	client
    .useXpath()
	 //Settings//
	.click('//*[@id="menu-content"]/li[4]')
	.pause(3000)
	 //Company Profile//
	.click('//*[@id="settings"]/li[2]')
	.pause(3000)
	 //Categories//
	.click('//*[@id="settings"]/li[3]')
	.pause(3000)
	 //Dashboards//
	.click('//*[@id="settings"]/li[4]')
	.pause(3000)
	 //Home//
	.click('//*[@id="navbar"]/ul[1]/li')
	.pause(5000)
};

exports.topicName1=function(client) {
	client
	  .useXpath()
	  .moveToElement(topicName1,10, 10)
	  .pause(3000)
};

exports.topicDeleteDelete=function(client) {
	client
	  .useXpath()
	  .click(topicDeleteButton)
	  .pause(3000)
	  .useCss()
	  .moveToElement(topicDeleteModel,10, 10)
	  .pause(3000)
	  .click(topicDeleteDeleteButton)
	  .pause(3000)
};

exports.end=function(client) {
	client
	  .end(); 
};
