// nightwatch/DiscoveryNow_WebUI_TestCases/tests/05_discoverNow.js
/*
 * DiscoveryNow Web UI
 * Login --> Create Topic --> Dataset Overview --> Alerts Settings --> Settings
 * 
 *
 * Attensity Group LLC (2015)
 */
 
	qcredentials = require('./qcredentials.json');
	qconfig = require('./qconfig.json');
		
	var qaURL = qconfig.qaURL;
	var loginButton = qconfig.loginButton;
	var newTopic = qconfig.newTopic;
	var sources = qconfig.sources;
	var allSources = qconfig.allSources;
	var doneButton = qconfig.doneButton;
	var saveTopic = qconfig.saveTopic;
	var home = qconfig.home;
	var topicName1 = qconfig.topicName1;
	var topicName2 = qconfig.topicName2;
	var topicDeleteButton = qconfig.topicDeleteButton;
	var topicDeleteModel = qconfig.topicDeleteModel;
	var topicDeleteCancelButton = qconfig.topicDeleteCancelButton;
	var topicDeleteDeleteButton = qconfig.topicDeleteDeleteButton;
	
exports.url=function(client) {
	client
		.url(qaURL)
		.pause(3000)
		.waitForElementVisible('body', 9000)
		.maximizeWindow()
		.pause(3000)
};

exports.login=function(client) {
	client
		.setValue('input#account', qcredentials.account)
		.setValue('input#login', qcredentials.login)
		.setValue('input#password', qcredentials.password)
		.pause(1000)
		.click(loginButton)
		.pause(3000)
};

exports.newTopic=function(client) {
	client
		.click(newTopic)
		.pause(3000)
		.setValue('input[type=text]', ['super'])
		.pause(3000)
		.keys([client.Keys.SPACE])
		.pause(2000)
};

exports.sources=function(client) {
	client
		.click(sources)
		.pause(2000)
};

exports.allSources=function(client) {
	client
		.click(allSources)
		.pause(2000)
		.click(doneButton)
		.pause(10000)
};

exports.saveTopic=function(client) {
	client
		.click(saveTopic)
		.pause(3000)
		.setValue('input[name=topicName]', ['AutomationWebUITopic'])
		.pause(3000)
		.keys([client.Keys.TAB])
		.keys([client.Keys.TAB])
		.keys([client.Keys.TAB])
		.keys([client.Keys.TAB])
		.keys([client.Keys.ENTER])
		.pause(10000)
};

exports.sentimentTrendWidget=function(client) {
	client
		.useXpath()
		 //Sentiment Trend DropDown//
		.click('/html/body/div[1]/div[2]/div/div/div[2]/div[2]/div/div[3]/div/div[2]/sentiment-trend-widget/div/div[1]/div[2]/a')
		.pause(3000)
		 //Export//
		.click('/html/body/div[1]/div[2]/div/div/div[2]/div[2]/div/div[3]/div/div[2]/sentiment-trend-widget/div/div[1]/div[2]/ul/li[1]/a')
		.pause(3000)
		//Sentiment Trend DropDown//
		.click('/html/body/div[1]/div[2]/div/div/div[2]/div[2]/div/div[3]/div/div[2]/sentiment-trend-widget/div/div[1]/div[2]/a')
		.pause(3000)
		 //Save Chart//
		.click('/html/body/div[1]/div[2]/div/div/div[2]/div[2]/div/div[3]/div/div[2]/sentiment-trend-widget/div/div[1]/div[2]/ul/li[2]/a/span/span')
		.pause(3000)
};

exports.bubbleChartWidget=function(client) {
	client
		.useXpath()
		//Top 50//
		.click('/html/body/div[1]/div[2]/div/div/div[2]/div[2]/div/div[3]/div/div[3]/top-entities-bubble-chart/div/div[2]/div[2]/div/a')
		.pause(3000)
		//Drop From Top 50//
		.click('/html/body/div[1]/div[2]/div/div/div[2]/div[2]/div/div[3]/div/div[3]/top-entities-bubble-chart/div/div[2]/div[2]/div/ul/li[1]/a')
		.pause(3000)
		//Top 50//
		.click('/html/body/div[1]/div[2]/div/div/div[2]/div[2]/div/div[3]/div/div[3]/top-entities-bubble-chart/div/div[2]/div[2]/div/a')
		.pause(3000)
		//Sentiment//
		.click('/html/body/div[1]/div[2]/div/div/div[2]/div[2]/div/div[3]/div/div[3]/top-entities-bubble-chart/div/div[2]/div[2]/div/div[1]/a[2]')
		.pause(3000)
		//Type//
		.click('/html/body/div[1]/div[2]/div/div/div[2]/div[2]/div/div[3]/div/div[3]/top-entities-bubble-chart/div/div[2]/div[2]/div/div[1]/a[3]')
		.pause(3000)
		//Terms of Interest//
		.click('/html/body/div[1]/div[2]/div/div/div[2]/div[2]/div/div[3]/div/div[3]/top-entities-bubble-chart/div/div[2]/div[2]/div/div[1]/a[1]')
		.pause(3000)
		 //Bubble Chart DropDown//
		.click('/html/body/div[1]/div[2]/div/div/div[2]/div[2]/div/div[3]/div/div[3]/top-entities-bubble-chart/div/div[1]/div[2]/a')
		.pause(3000)
		 //Export//
		.click('/html/body/div[1]/div[2]/div/div/div[2]/div[2]/div/div[3]/div/div[3]/top-entities-bubble-chart/div/div[1]/div[2]/ul/li[1]/a')
		.pause(3000)
		//Bubble Chart DropDown//
		.click('/html/body/div[1]/div[2]/div/div/div[2]/div[2]/div/div[3]/div/div[3]/top-entities-bubble-chart/div/div[1]/div[2]/a')
		.pause(3000)
		//Save Chart//
		.click('/html/body/div[1]/div[2]/div/div/div[2]/div[2]/div/div[3]/div/div[3]/top-entities-bubble-chart/div/div[1]/div[2]/ul/li[2]/a/span/span')
		.pause(3000)
		//Bubble Chart DropDown//
		.click('/html/body/div[1]/div[2]/div/div/div[2]/div[2]/div/div[3]/div/div[3]/top-entities-bubble-chart/div/div[1]/div[2]/a')
		.pause(3000)
		//Hidden Values//
		.click('/html/body/div[1]/div[2]/div/div/div[2]/div[2]/div/div[3]/div/div[3]/top-entities-bubble-chart/div/div[1]/div[2]/ul/li[3]/a')
		.pause(3000)
		//Hidden Values Close//
		.click('/html/body/div[15]/div/div/div[3]/button')
		.pause(3000)
};

exports.sourcesWidget=function(client) {
	client
		.useXpath()
		 //Sources DropDown//
		.click('/html/body/div[1]/div[2]/div/div/div[2]/div[2]/div/div[3]/div/div[4]/social-channels-widget/div/div[1]/div[2]/a')
		.pause(3000)
		 //Export//
		.click('/html/body/div[1]/div[2]/div/div/div[2]/div[2]/div/div[3]/div/div[4]/social-channels-widget/div/div[1]/div[2]/ul/li[1]/a')
		.pause(3000)
		//Sources DropDown//
		.click('/html/body/div[1]/div[2]/div/div/div[2]/div[2]/div/div[3]/div/div[4]/social-channels-widget/div/div[1]/div[2]/a')
		.pause(3000)
		//Save Chart//
		.click('/html/body/div[1]/div[2]/div/div/div[2]/div[2]/div/div[3]/div/div[4]/social-channels-widget/div/div[1]/div[2]/ul/li[2]/a/span/span')
		.pause(3000)
		 //Percentage//
		.click('/html/body/div[1]/div[2]/div/div/div[2]/div[2]/div/div[3]/div/div[4]/social-channels-widget/div/div[2]/div[1]/div[2]/div/a[2]')
		.pause(3000)
		 //Number of Mentions//
		.click('/html/body/div[1]/div[2]/div/div/div[2]/div[2]/div/div[3]/div/div[4]/social-channels-widget/div/div[2]/div[1]/div[2]/div/a[1]')
		.pause(3000)
};

exports.globalWidget=function(client) {
	client
		.useXpath()
		 //Global DropDown//
		.click('/html/body/div[1]/div[2]/div/div/div[2]/div[2]/div/div[3]/div/div[5]/heat-map-widget/div[1]/div[2]/span')
		.pause(3000)
		 //Export//
		.click('/html/body/div[1]/div[2]/div/div/div[2]/div[2]/div/div[3]/div/div[5]/heat-map-widget/div[1]/div[2]/ul/li[1]/a')
		.pause(3000)
		 //Global DropDown//
		.click('/html/body/div[1]/div[2]/div/div/div[2]/div[2]/div/div[3]/div/div[5]/heat-map-widget/div[1]/div[2]/span')
		.pause(3000)
		 //Save Chart//
		.click('/html/body/div[1]/div[2]/div/div/div[2]/div[2]/div/div[3]/div/div[5]/heat-map-widget/div[1]/div[2]/ul/li[2]/a')
		.pause(3000)
};

exports.end=function(client) {
	client
	  .end();
};
